src/headers_ext/content_coding.rs

```diff
@@ -0,0 +1,154 @@
+            #[inline]
+            pub fn to_static(&self) -> &'static str {
+                match *self {
+                    $(ContentCoding::$coding => $str,)+
+                }
+            }
+        }
+
+        impl From<&str> for ContentCoding {
+            /// Given a `&str` returns a `ContentCoding`
+            ///
+            /// Note this will never fail, in the case of `&str` being an invalid content coding,
+            /// will return `ContentCoding::IDENTITY` because `'identity'` is generally always an
+            /// accepted coding.
+            ///
+            /// # Example
+            ///
+            /// ```
+            /// use static_web_server::headers_ext::ContentCoding;
+            ///
+            /// let invalid = ContentCoding::from("not a valid coding");
+            /// assert_eq!(invalid, ContentCoding::IDENTITY);
+            ///
+            /// let valid = ContentCoding::from("gzip");
+            /// assert_eq!(valid, ContentCoding::GZIP);
+            /// ```
+            #[inline]
+            fn from(s: &str) -> Self {
+                ContentCoding::from_str(s).unwrap_or_else(|_| ContentCoding::IDENTITY)
+            }
+        }
+
+        impl FromStr for ContentCoding {
+            type Err = ();
+
+            /// Given a `&str` will try to return a `ContentCoding`
+            ///
+            /// Different from `ContentCoding::from_str(&str)`, if `&str` is an invalid content
+            /// coding, it will return `Err(())`
+            ///
+            /// # Example
+            ///
+            /// ```
+            /// use static_web_server::headers_ext::ContentCoding;
+            /// use std::str::FromStr;
+            ///
+            /// let invalid = ContentCoding::from_str("not a valid coding");
+            /// assert!(invalid.is_err());
+            ///
+            /// let valid = ContentCoding::from_str("gzip");
+            /// assert_eq!(valid.unwrap(), ContentCoding::GZIP);
+            /// ```
+            #[inline]
+            fn from_str(s: &str) -> Result<Self, Self::Err> {
+                match s {
+                    $(
+                        stringify!($coding)
+                        | $str => Ok(Self::$coding),
+                    )+
+                    _ => Err(())
+                }
+            }
+        }
+
+        impl std::fmt::Display for ContentCoding {
+            #[inline]
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+                write!(f, "{}", match *self {
+                    $(ContentCoding::$coding => $str.to_string(),)+
+                })
+            }
+        }
+
+        impl From<ContentCoding> for HeaderValue {
+            fn from(coding: ContentCoding) -> HeaderValue {
+                match coding {
+                    $(ContentCoding::$coding => HeaderValue::from_static($str),)+
+                }
+            }
+        }
+    }
+}
+
+define_content_coding! {
+    ANY; "*",
```

Comment:
> I’ve also added `ContentCoding::ANY` (corresponds to `Accept-Encoding: *`). This is a valid value which probably needs to be considered by SWS at some point.

This seems to me like a new feature for SWS. Can we please move the code (also tests) around this out to a new PR? Just for increase clarity.